advanced/Scripts/updatecheck.sh

```diff
@@ -37,6 +37,8 @@ rm -f "/etc/pihole/localversions"
 VERSION_FILE="/etc/pihole/versions"
 touch "${VERSION_FILE}"
 chmod 644 "${VERSION_FILE}"
+# if /pihole.docker.tag file exists, we will use it's value later in this script
+DOCKER_TAG=$(cat file 2>/dev/null)
```

Comment:
Is there anything in the function that will prevent unexpected data that is read in from that file?

Reply:
I guess no more than there would have been if anyone had passed in unexpected data to `PIHOLE_DOCKER_TAG` env var. I guess expected values are `nightly`, `dev`, or version number in the format `YYYY.MM[.n]`, so we could test for that, but I wouldn't want to rule out the tag name being something arbitrary, either (in the case of extra special containers, `beta`, `experiment`, etc)

Reply:
Are you worried about code execution or just the wrong text?

Reply:
I'm worried about anything and everything when we read a file in.

Reply:
This file is just replacing the environment variable. Anything you can add into this file you could add to the var. 
I think the security implications are exactly the same.

Reply:
Good point. We should take care code execution is not happening at the web interface here
https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/update_checker.php

or https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/footer.php

if arbitrary code is injected.

Reply:
I created a PR (web interface) to filter values read from `/etc/pihole/versions` file.

> ... I wouldn't want to rule out the tag name being something arbitrary, either (in the case of extra special containers, beta, experiment, etc)

My code allows: Uppercase and lowercase letters, digits, dot, colon, dash, underscore and slash. I think this way we will be able to use any tag we want.

Reply:
I have something for this side... HMB

Reply:
![image](https://user-images.githubusercontent.com/1998970/192092596-9a3ea870-0b59-4aa6-b00b-ceb5d5d7fcf5.png)